opnode/cmd/main.go

```diff
@@ -1,31 +1,211 @@
 package main
 
 import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/urfave/cli"
+
 	"github.com/ethereum-optimism/optimistic-specs/opnode/node"
-	"github.com/protolambda/ask"
 )
 
-type MainCmd struct {
+var (
+	Version   = ""
+	GitCommit = ""
+	GitDate   = ""
+)
+
+func main() {
+	// Set up logger with a default INFO level in case we fail to parse flags,
+	// otherwise the final critical log won't show what the parsing error was.
+	log.Root().SetHandler(
+		log.LvlFilterHandler(
+			log.LvlInfo,
+			log.StreamHandler(os.Stdout, log.TerminalFormat(true)),
+		),
+	)
+
+	app := cli.NewApp()
+	app.Flags = Flags
+	app.Version = fmt.Sprintf("%s-%s", Version, params.VersionWithCommit(GitCommit, GitDate)) // TODO: Don't use Geth's hardcoded versions
+	app.Name = "opnode"
+	app.Usage = "Optimism Rollup Node"
+	app.Description = "The deposit only rollup node drives the L2 execution engine based on L1 deposits."
+
+	app.Action = RollupNodeMain
+	err := app.Run(os.Args)
+	if err != nil {
+		log.Crit("Application failed", "message", err)
+	}
+}
+
+func RollupNodeMain(ctx *cli.Context) error {
+	log.Info("Initializing Rollup Node")
+	cfg, err := NewConfig(ctx)
+	if err != nil {
+		log.Error("Unable to create the rollup node config", "error", err)
+		return err
+	}
+	n, err := node.New(context.Background(), cfg)
```

Comment:
I think we should create the logger from the log-config outside of the `node.New`, and insert it as argument. That way we can pass test-loggers, and tag node instances with different context params to differentiate between the logs in a multi-opnode test.